crates/engine/tree/src/tree/payload_processor/chunking.rs

```diff
@@ -0,0 +1,80 @@
+//! Generic chunk planning utilities for multiproof scheduling in the engine.
+
+/// A computed plan for splitting `total` items into `chunks` parts.
+///
+/// Sizes are distributed as evenly as possible:
+/// - `base` items per chunk
+/// - the first `increased` chunks get one extra item (size `base + 1`)
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub(super) struct ChunkPlan {
```

Comment:
instead of having this as a separate file, which creates complexity and additional code overhead to manage, i think it would be better to inline it in areas where we would need it.